application/src/main/kotlin/net/corda/v5/application/uniqueness/model/UniquenessCheckResult.kt

```diff
@@ -0,0 +1,130 @@
+package net.corda.v5.application.uniqueness.model
+
+import net.corda.data.uniqueness.UniquenessCheckExternalResponse
+import net.corda.data.uniqueness.UniquenessCheckExternalResultInputStateConflict
+import net.corda.data.uniqueness.UniquenessCheckExternalResultInputStateUnknown
+import net.corda.data.uniqueness.UniquenessCheckExternalResultMalformedRequest
+import net.corda.data.uniqueness.UniquenessCheckExternalResultReferenceStateConflict
+import net.corda.data.uniqueness.UniquenessCheckExternalResultReferenceStateUnknown
+import net.corda.data.uniqueness.UniquenessCheckExternalResultTimeWindowOutOfBounds
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import org.apache.avro.specific.SpecificRecord
+import java.time.Instant
+
+/**
+ * TODO Rewrite KDocs
+ */
+sealed class UniquenessCheckResult(val commitTimestamp: Instant) {
+    companion object {
+        const val RESULT_ACCEPTED_REPRESENTATION = 'A'
+        const val RESULT_REJECTED_REPRESENTATION = 'R'
+
+        /**
+         * Converts an Avro error to a [Failure]. Since there's no common superclass
+         * for avro classes, the parameter can be anything, hence the [Any] type.
+         */
+        fun fromExternalError(avroResponse: UniquenessCheckExternalResponse): Failure {
```

Comment:
Given this is taking a response (not specifically an error response) wouldn't it make more sense to call this `fromAvroResponse` and have it handle the success case too?